app/adapters/telegram_adapter/inbound.rb

```diff
@@ -48,6 +50,17 @@ def consume(telegram_message)
       yield(@message) if block_given?
     end
 
+    def avatar_url(contributor)
+      return unless contributor.telegram_id
+
+      profile_photos = Telegram.bot.get_user_profile_photos(user_id: contributor.telegram_id, limit: 1).with_indifferent_access
+      first_photo = profile_photos.dig(:result, :photos, 0)
+      return unless first_photo
+
+      largest_size = first_photo.max { |a, b| a[:file_size] <=> b[:file_size] }
```

Comment:
I was wondering if we might want to reduce duplication of this line (we select the highest resolution for images attached to messages as well as profile photos) and extract it into something like `photo_url`? On the other hand, it’s just a single line duplicated in two places, so definitely nothing super important.

Reply:
Yep, I thought about putting that line in a central location but decided against it.